src/crystal/main.cr

```diff
@@ -56,7 +56,9 @@ module Crystal
         1
       end
 
-    AtExitHandlers.run status
+    new_status = AtExitHandlers.run status
```

Comment:
Surely there would be a lot less duplication if this main function simply called `exit` with the `status` and never returned.

Reply:
Good idea, I'll do that

Reply:
But this will make `main` never return, that's probably not what we want IMO, it'll probably break some other things if we do that.

@ysbaddaden do you know something about this?
I'd like to have input from @asterite too (let's hope :smiley:)